src/Microsoft.DotNet.Build.Tasks/GenerateBindingRedirect.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Task = Microsoft.Build.Utilities.Task;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateBindingRedirect : Task
+    {
+        [Required]
+        public ITaskItem[] AssemblyIdentities { get; set; }
+
+        [Required]
+        public ITaskItem[] Executables { get; set; }
+
+        [Required]
+        public string OutputPath { get; set; }
+
+        private static XNamespace ns { get; set; }
+
+        public override bool Execute()
+        {
+            ns = "urn:schemas-microsoft-com:asm.v1";
+            XElement bindingRedirectAssemblies = new XElement(ns + "assemblyBinding");
+            List<XElement> dependentAssemblyList = new List<XElement>();
+            foreach (ITaskItem assembly in AssemblyIdentities)
+            {
+                string publicKeyToken = assembly.GetMetadata("PublicKeyToken");
+                if (!string.IsNullOrEmpty(publicKeyToken))
+                {
+                    string assemblyVersion = assembly.GetMetadata("Version");
+                    string assemblyName = assembly.GetMetadata("Name");
+                    string culture = string.IsNullOrEmpty(assembly.GetMetadata("Culture"))
+                        ? "neutral"
+                        : assembly.GetMetadata("Culture");
+                    XElement assemblyIdentity = new XElement(ns + "assemblyIdentity",
+                        new XAttribute("name", assemblyName),
+                        new XAttribute("publicKeyToken", publicKeyToken),
+                        new XAttribute("culture", culture));
+                    XElement bindingRedirect = new XElement(ns + "bindingRedirect",
+                        new XAttribute("oldVersion", $"0.0.0.0-{assemblyVersion}"),
+                        new XAttribute("newVersion", assemblyVersion));
+                    XElement dependentAssembly = new XElement(ns + "dependentAssembly",
+                        assemblyIdentity,
+                        bindingRedirect);
+                    dependentAssemblyList.Add(dependentAssembly);
+                }
+            }
+            foreach (XElement element in dependentAssemblyList)
```

Comment:
Why do you need this additional `foreach`? Why not directly add them to `bindingRedirectAssemblies` ?